securityscanutils/securityscan.go

```diff
@@ -126,6 +130,14 @@ func (s *SecurityScanner) GenerateSecurityScans(ctx context.Context) error {
 		os.Remove(sarifTplFile)
 	}()
 
+	//Recreate a .trivyignore file locally
+	ignoreFile, err := ioutil.TempFile("", "")
+	if err != nil {
+		return err
+	}
+	ignoreFile.WriteString(s.TrivyIgnoreContents)
+	defer os.Remove(ignoreFile.Name())
+
```

Comment:
I think this leaks information about trivy scanning into the broader scanner. Ideally, we should isolate all trivy logic into the TrivyScanner struct. Instead of passing this file around, could we instantiate the trivyScanner with the list of cves to ignore, or call RunTrivyScan and pass the list of cves? That way we can keep the fact that it's a file (and where that file lives) hidden from the rest of the components.